src/gc/gc.cpp

```diff
@@ -34161,7 +34163,7 @@ GCHeap::Alloc( size_t size, uint32_t flags REQD_ALIGN_DCL)
 
     TRIGGERSGC();
 
-    assert (!GCHeap::UseAllocationContexts());
+    assert (!UseAllocationContexts());
```

Comment:
These asserts should be removed and the implementation of `UseAllocationContexts` deleted from gc.h because of it is duplicated with vm. The logic to decide whether to use AllocationContexts should be in VM - it does not need to be duplicated in the GC.